runtime/lua/vim/shared.lua

```diff
@@ -62,7 +62,7 @@ end)()
 ---
 ---@param s string String to split
 ---@param sep string Separator or pattern
----@param plain boolean If `true` use `sep` literally (passed to string.find)
+---@param plain? boolean If `true` use `sep` literally (passed to string.find)
```

Comment:
```suggestion
---@param (plain|nil) boolean If `true` use `sep` literally (passed to string.find)
```
(That's the convention we've adopted throughout the codebase for optional parameters.)

Reply:
Maybe it's off. That is not the correct way to write annotations. It's this way, right?

Reply:
Both are correct according to EmmyLua, but we chose the one I gave because it works with our documentation generator.

Reply:
Uh, `plain` is a variable name, so you're writing it that way incorrectly. Check your writing carefully.

Reply:
Also, there are several `@param variable?` notations in shared.lua, should I modify this to `type|nil`?

Reply:
> Uh, `plain` is a variable name, so you're writing it that way incorrectly. Check your writing carefully. Is this what you mean?
> 
> ```lua
> ---@param plain (boolean|nil) If `true` use `sep` literally (passed to string.find)
> ```

Yes, sorry, that was obviously a typo.

> Also, there are several @param variable? notations in shared.lua, should I modify these to type|nil?

Please don't make unrelated changes in a single PR, that makes it very hard to review and merge.

Reply:
This PR is intended to correct annotations.

Reply:
Then I don't understand the functional changes to the Lua code...

Reply:
Is it a change in vim.pesc?
It is a fix for the part that does not match the annotation and LS send diagnostics.
If you put string.gsub as it is in return, it will return multiple values. There may be no practical problem, but it is not good to have a difference between the annotation and the actual behavior.
```lua
-- Example
local function foo()
    return string.gsub('abc', 'a', 'b')
end
vim.pretty_print(foo())
```

Reply:
The same reason why nil in vim.tbl_islist is changed to false, nil is not a boolean. If nil should be returned here, modify the annotation to boolean|nil.